File: src/test/java/cn/edu/cug/cs/gtl/series/ml/SeriesTest.java
package cn.edu.cug.cs.gtl.series.ml;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class SeriesTest {

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void of() throws  Exception{
        double [] doubles = new double[10];
        for(int i=0;i<10;i++)
            doubles[i]=i;
        Series s = new Series(doubles);
        Series s1 = (Series) s.clone();
        byte [] bs = s.storeToByteArray();
        byte [] bs1 = s1.storeToByteArray();
        Assert.assertArrayEquals(bs,bs1);
    }

    @Test
    public void testOf() {
    }

    @Test
    public void testOf1() {
    }

    @Test
    public void testOf2() {
    }

    @Test
    public void testOf3() {
    }

    @Test
    public void load() {
    }

    @Test
    public void store() {
    }

    @Test
    public void compareTo() {
    }
}

File: src/test/kotlin/cn/edu/cug/cs/gtl/series/SeriesTest.kt
package cn.edu.cug.cs.gtl.series

import cn.edu.cug.cs.gtl.config.Config
import cn.edu.cug.cs.gtl.io.File
import cn.edu.cug.cs.gtl.series.common.Series
import org.junit.Assert
import org.junit.Test
import java.io.*

class SeriesTest {
    @Test
    fun copy() {
        val ys = doubleArrayOf(50.0, 10.0, 20.0, 30.0, 40.0, 70.0, 90.0, 10.0, 30.0, 40.0)
        val s1 = Series.of(ys)
        try {
            val f = FileOutputStream(testFileName)
            s1.write(f)
            f.close()
            val f2 = FileInputStream(testFileName)
            val s3 = Series.of(f2)
            Assert.assertArrayEquals(ys, s3.values, 0.001)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    @Test
    fun load() {
        val ys = doubleArrayOf(50.0, 10.0, 20.0, 30.0, 40.0, 70.0, 90.0, 10.0, 30.0, 40.0)
        val s1 = Series.of(ys)
        try {
            val baos = ByteArrayOutputStream()
            s1.write(baos)
            val bytes = baos.toByteArray()
            baos.close()
            val bais = ByteArrayInputStream(bytes)
            val s3 = Series.of(bais)
            Assert.assertArrayEquals(ys, s3.values, 0.001)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    @Test
    fun store() {
        val ys = doubleArrayOf(50.0, 10.0, 20.0, 30.0, 40.0, 70.0, 90.0, 10.0, 30.0, 40.0)
        val s1 = Series.of(ys)
        try {
            val bytes = s1.storeToByteArray()
            val s3 = Series.of(bytes)
            Assert.assertArrayEquals(ys, s3.values, 0.001)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    companion object {
        var testFileName = (Config.getTestOutputDirectory()
                + File.separator + "test.series")
    }
}